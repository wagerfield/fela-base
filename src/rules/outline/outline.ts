export interface OutlineRuleOptions {
  color?: string
  style?: string
  width?: string
  offset?: string
}

export default ({
  color = "initial",
  style = "initial",
  width = "initial",
  offset = "initial"
}: OutlineRuleOptions = {}) => ({
  outlineColor: color,
  outlineStyle: style,
  outlineWidth: width,
  outlineOffset: offset
})
