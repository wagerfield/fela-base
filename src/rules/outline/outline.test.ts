import outline from "./outline"

describe("outline(props)", () => {
  it("renders expected snapshots", () => {
    expect(outline()).toMatchSnapshot()
    expect(
      outline({
        width: "thick"
      })
    ).toMatchSnapshot()
  })
})
