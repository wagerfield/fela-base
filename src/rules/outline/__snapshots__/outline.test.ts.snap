// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outline(props) renders expected snapshots 1`] = `
Object {
  "outlineColor": "initial",
  "outlineOffset": "initial",
  "outlineStyle": "initial",
  "outlineWidth": "initial",
}
`;

exports[`outline(props) renders expected snapshots 2`] = `
Object {
  "outlineColor": "initial",
  "outlineOffset": "initial",
  "outlineStyle": "initial",
  "outlineWidth": "thick",
}
`;
